ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx

```diff
@@ -0,0 +1,197 @@
+      setData(rootNode);
+    }
+  }, [rootItem]);
+
+  useEffect(() => {
+    const nextFetching = [];
+    fetchingPathsRef.current.forEach((path) => {
+      if (childrenByParentPath[path]) {
+        nodesByPathRef.current[path].children = [];
+        childrenByParentPath[path].forEach((childPath) => {
+          const node = {
+            id: childPath,
+            name: itemsByPath[childPath].label,
+            children: [{ id: 'loading' }]
+          };
+          nodesByPathRef.current[path].children.push(node);
+          nodesByPathRef.current[childPath] = node;
+        });
+
+        setData({ ...nodesByPathRef.current[rootPath] });
+      } else {
+        nextFetching.push(path);
+      }
+    });
+    fetchingPathsRef.current = nextFetching;
+  }, [data, rootPath, childrenByParentPath]);
+
+  const onChangeCollapsed = (collapsed) => {};
+
+  const onLabelClick = (path: string) => {
+    if (isNavigable(itemsByPath[path])) {
+      console.log('preview: ', path);
+    } else {
+      onIconClick(path);
+    }
+  };
+
+  const onIconClick = (path: string) => {
+    // If the path is already expanded, pathNavigatorTreeUpdate should collapsed it
+    if (state.expanded.includes(path)) {
+      dispatch(
+        pathNavigatorTreeUpdate({
+          id,
+          expanded: state.expanded.filter((expanded) => expanded !== path)
+        })
+      );
+    } else {
+      // if: the path does not exist on childrenByParentPath, the children was not fetched
+      // else: the tree should expand the selected node
+      if (!childrenByParentPath[path]) {
+        fetchingPathsRef.current.push(path);
+        dispatch(
+          pathNavigatorTreeFetchPathChildren({
+            id,
+            path
+          })
+        );
+      } else {
+        dispatch(
+          pathNavigatorTreeUpdate({
+            id,
+            expanded: [...state.expanded, path]
+          })
+        );
+      }
+    }
+  };
+
+  const onOpenItemMenu = (element: Element, path: string) => {
+    setItemMenu({
+      path,
+      anchorEl: element,
+      loaderItems: getNumOfMenuOptionsForItem(itemsByPath[path])
+    });
+  };
+
+  const onCloseItemMenu = () => setItemMenu({ ...itemMenu, path: null, anchorEl: null });
+
+  return (
+    <ConditionalLoadingState isLoading={!rootItem || !Boolean(state) || !data}>
+      <PathNavigatorTreeUI
+        title={label}
+        icon={icon}
+        container={container}
+        isCollapsed={state?.collapsed}
+        data={data}
+        itemsByPath={itemsByPath}
+        expandedNodes={state?.expanded}
+        onIconClick={onIconClick}
+        onLabelClick={onLabelClick}
+        onChangeCollapsed={onChangeCollapsed}
+        onOpenItemMenu={onOpenItemMenu}
+      />
+      <ItemActionsMenu
+        open={Boolean(itemMenu.anchorEl)}
+        path={itemMenu.path}
+        numOfLoaderItems={itemMenu.loaderItems}
+        anchorEl={itemMenu.anchorEl}
+        onClose={onCloseItemMenu}
+      />
```

Comment:
Should we use the GlobalDialogManager item menu?

Reply:
we are using this approach on PathNavigatorTree, the issue using the one on GlobalDialogManager is that we need to send anchor position

Reply:
Let's try/discuss something like this...
```typescript
import { getOffsetTop, getOffsetLeft } from '@material-ui/core/Popover'

onOpenMenu = (e) => {
  const anchorRect = e.target.getBoundingClientRect();
  return {
    top: anchorRect.top + getOffsetTop(anchorRect, 'top'),
    left: anchorRect.left + getOffsetLeft(anchorRect, 'left'),
  }
}

```